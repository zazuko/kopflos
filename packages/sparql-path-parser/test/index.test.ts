import { expect, use } from 'chai'
import { jestSnapshotPlugin } from 'mocha-chai-jest-snapshot'
import { toSparql } from 'clownface-shacl-path'
import { parse } from '../src/index.js'

describe('sparql-path-parser', () => {
  use(jestSnapshotPlugin())

  describe('parse', () => {
    [
      'schema:foo / schema:bar', // test whitespace is ignored
      '!(rdf:type|^rdf:type)',
    ]
      .forEach((path) => {
        it(`'${path}'`, () => {
          expect(toSparql(parse(path)).toString({ prologue: false })).toMatchSnapshot()
        })
      })
  })
})
