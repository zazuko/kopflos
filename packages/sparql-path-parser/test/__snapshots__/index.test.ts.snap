// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sparql-path-parser parse '!(rdf:type|^rdf:type)' 1`] = `"!(rdf:type|^rdf:type)"`;

exports[`sparql-path-parser parse 'schema:foo / schema:bar' 1`] = `"schema:foo/schema:bar"`;
